pkg/api/types.go

```diff
@@ -1003,6 +1003,16 @@ type NodeSpec struct {
 
 	// Unschedulable controls node schedulability of new pods. By default node is schedulable.
 	Unschedulable bool `json:"unschedulable,omitempty"`
+
+	// Certificate is the public x.509 certificate tied to the node contained
+	// in a PEM-encoded byte array. It is presented by the node when it asks to
+	// join the cluster and can be refreshed if the certificate on the node is
+	// rotated. The node certificate is checked when initiating a connection to
+	// the node (to verify that the connection isn't being intercepted) and is
+	// presented by the node when it connects to the master (as a client
+	// certificate) to establish that the connection is coming from a trusted
+	// and identifiable source.
+	Certificate []byte `json:certificate,omitempty"`
 }
```

Comment:
+1 to what @erictune says. I can really only see storing the pub key here if you're using public keys in lieu of passwords (i.e. to establish identity). If you're just using it for authz/n, just a cert chain should be sufficient?

Actually maybe not even for identity-- if they have a cert signed by something we trust, we also trust them to handle their cert appropriately by e.g. not sharing it or lying to us about their identity...

Sanity checking that every node presents a unique cert sounds like it should be part of a generalized cert rotation strategy...